### ISSUE TITLE
RFC: Thoughts on Black?

### ISSUE THREAD

**Issue by itsthejoker** [GrafeasGroup/tor#127]
https://github.com/ambv/black

This has crossed my dashboard more than once recently and every time it does, it looks more interesting. I'm curious to know what thoughts people have on adopting the Black code style for our projects.

**Comment by itsthejoker** [comment]
Comment from @TheLonelyGhost: "I'd have to toy with it a bit, but it looks like something very useful if for no other reason than uniformity of the source code."

I'll run Black against the current codebase and open it as a PR so we can see what we're looking at.

**Comment by TheLonelyGhost** [comment]
Yeah, I don’t mind the changes and it takes the thinking of how to best format the code completely out of the equation. I really like that.

The only thing that throws me off is the emphasis on `”` instead of `’` for strings. Other languages use `’` strings to protect from string interpolation and character escapes like `\n`. That’s not so much the case in python, I realize, and it will likely just be an adjustment period for me.

All things said, I’m in favor of it. We just need to map out using black with Flake8 or PyLint. Plus running that stuff as part of TravisCI or other CI pipeline is an item on my to do list. Hosted style solutions like Codacy just aren’t cutting it for me.

**Comment by itsthejoker** [comment]
I agree -- this is the section on their justification for it: https://black.readthedocs.io/en/stable/the_black_code_style.html#strings

**Comment by DavidArchibald** [comment]
I've been using Black for a while now, largely inspired by this issue. I did not comment on it then because I had had no experience in it. However, I'd now like to report that since then I've come across no issues with using it in my personal projects. I've quite enjoyed how it formats everything for me, eliminating some of the more tedious, repetitive work. Even better, Black is fast enough that I was able to hook it up to format the file every time I save it, so I don't have to stare at ugly code for too long, or switch from writing code to run it. I think adding it to the Repo, or really any style guide will be beneficial because it will prevent styles from conflicting. I opened #145 for some of the small style problems I noticed to alleviate this, but Black should _eliminate_ this problem entirely.
